File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (s.isBlank())
            return true;

        int i = 0;
        int j = 0;
        int sublen = s.length();
        int suplen = t.length();
        String format = "%-40s%s%n";
        char letter = s.charAt(i);
        while (i<sublen && j<suplen){
            if (s.charAt(i)==t.charAt(j))
                i++;
            j++;
        }
//        for (int j = 0; j < t.length() && i < s.length(); j++) {
//            System.out.println("I: "+i+" "+s.charAt(i)+"\tJ: "+j +" "+t.charAt(j));
//            if (s.charAt(i)==t.charAt(j)){
//                if (i==s.length()-1)
//                    return true;
//                i++;}
//        }
//        for (char c :
//                t.toCharArray()) {
//            if (i==s.length()-1)
//                return true;
//            if (letter == c){
//                i++;
//                letter = s.charAt(i);
//            }
//        }
        if (i==sublen)
            return true;
        else return false;

    }
}

File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        l = [ch for ch in t if ch in s]
        i, j = 0, 0
        while i < len(s) and j < len(l):
            if s[i] == l[j]:
                i += 1
            j += 1
        return i == len(s)
        # return s in "".join([ch for ch in t if ch in s]).replace(" ", "")